### DISCUSSION TITLE
Berquist Sherman Adjustment on Paid Loss Triangles

### DISCUSSION THREAD

**Discussion by IGotCavities** [casact/chainladder-python#417] [Q&A] [answered]
Was playing around to learn ChainLadder but couldn't seem to find the Paid B/S method that adjusts the paid triangle through disposal rate. Does anyone know if that method exists within ChainLadder package? Can you point me there? 

I have seen and gone through the (incurred) Berquist Sherman method that adjusts triangles based on a predetermined trend assumption, which is pretty awesome! 

Thanks!

**Comment by jbogaardt** [comment]
Its been a while since I've looked at `BerquistSherman`, but the [docs](https://chainladder-python.readthedocs.io/en/latest/adjustments.html#berquistsherman) do describe the `disposal_rate_` property of the esimtator very briefly. I think  that if you supply a paid, case incurred, reported count, and closed count triangle, the method should automatically adjust the paids for the disposal rate...i think.

**Reply by IGotCavities** [reply]
Thanks for the help! Got the initial adjusted paid triangles and am definitely interested in exploring more.

**Reply by IGotCavities** [reply]
Coming back for more Berquist question if I may. Want to specify the disposal rate to use instead of going for the one based on the default ultimate claim count. I found the "reported_count_estimate" parameter but not sure if I understand what it wants to be fed in. I found the explanation below in the source code, but too rookie to understand what "Estimate" or "Pipeline" means? Can you help understand and perhaps provide an example of using 3-yr volume weighted average to calculate disposal rate? Thanks! 

Code and the question is with the reported_count_estimator
berq_paid = cl.BerquistSherman(
    paid_amount='paidloss', incurred_amount='incdloss',
    reported_count='incdcount', closed_count='cwpcount',
    reported_count_estimator = (cl.Development(n_periods=3, average = "volume").fit(CW_tri['incdcount']).cdf_ * CW_tri['incdcount'])
)
berq_paid

Source Code explanation: 
reported_count_estimator: Estimator
        An IBNR estimator for reported_count used to calculate closed_count
        disposal rates.  Estimator can be a Pipeline.  If None selected then
        basic Chainladder model is used."

**Comment by jbogaardt** [comment, marked as answer]
There is probably opportunity to improve the docs here.  Estimator generically refers to any sklearn-like modeling feature in the package. However, in this case, it is limited to estimators that spin off an `ultimate_`  property.  By default, the volume-weighted `Chainladder` is used, but it could be any other IBNR estimator. If you want to tailor the development patterns of the `Chainladder` you need to use a `Pipeline`. A working example with a 3-year volume weighted average reported count ultimate would look like this:    
```python
import chainladder as cl
triangle = cl.load_sample('berqsherm').loc['MedMal']

reported_count_estimator = cl.Pipeline([
  ('dev', cl.Development(n_periods=3, average="volume")),
  ('model', cl.Chainladder()),]
)

berq = cl.BerquistSherman(
    paid_amount='Paid', 
    incurred_amount='Incurred',
    reported_count='Reported', 
    closed_count='Closed',
    reported_count_estimator=reported_count_estimator,
    trend=0.15).fit(triangle)

berq.adjusted_triangle_
```

**Reply by IGotCavities** [reply]
Thanks for the explanation and example! Pipeline comes in very handy, and the paid B/S adjustment method works well for me now.
